File: testdata/jennies/builders/builder_delegation_in_disjunction/PHPBuilder/src/BuilderDelegationInDisjunction/ExternalLinkBuilder.php
<?php

namespace Grafana\Foundation\BuilderDelegationInDisjunction;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\BuilderDelegationInDisjunction\ExternalLink>
 */
class ExternalLinkBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\BuilderDelegationInDisjunction\ExternalLink $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\BuilderDelegationInDisjunction\ExternalLink();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\BuilderDelegationInDisjunction\ExternalLink
     */
    public function build()
    {
        return $this->internal;
    }

    public function url(string $url): static
    {
        $this->internal->url = $url;
    
        return $this;
    }

}

File: testdata/jennies/builders/builder_delegation_in_disjunction/JavaBuilder/builder_delegation_in_disjunction/ExternalLinkBuilder.java
package builder_delegation_in_disjunction;


public class ExternalLinkBuilder implements cog.Builder<ExternalLink> {
    protected final ExternalLink internal;
    
    public ExternalLinkBuilder() {
        this.internal = new ExternalLink();
    }
    public ExternalLinkBuilder url(String url) {
        this.internal.url = url;
        return this;
    }
    public ExternalLink build() {
        return this.internal;
    }
}
